src/components/editor-panels.js

```diff
@@ -2,30 +2,40 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Spinner, CheckboxControl, PanelBody, PanelRow } from '@wordpress/components';
+import { BaseControl, CheckboxControl, PanelBody, Spinner } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
 
-const CheckboxesGroup = ( { options, values, onChange } ) => {
+const CheckboxesGroup = ( { options, values, onChange, defaultRequiredSlug } ) => {
 	if ( ! Array.isArray( options ) ) {
 		return <Spinner />;
 	}
-	return options.map( ( { name, slug } ) => (
-		<PanelRow key={ slug }>
+	return options.map( ( { name, slug } ) => {
+		const isDefaultType = slug === defaultRequiredSlug;
+		const otherTypesSelected = values.length > 1 || ( values.length === 1 && values[0] !== defaultRequiredSlug );
+		const isDisabled = isDefaultType && !otherTypesSelected;
+
+		return (
 			<CheckboxControl
 				label={ name }
 				checked={ values.indexOf( slug ) > -1 }
+				disabled={ isDisabled }
 				onChange={ value => {
 					const cleanPostType = [ ...new Set( values ) ];
 					if ( value && cleanPostType.indexOf( slug ) === -1 ) {
 						cleanPostType.push( slug );
 					} else if ( ! value && cleanPostType.indexOf( slug ) > -1 ) {
 						cleanPostType.splice( cleanPostType.indexOf( slug ), 1 );
 					}
+					// If no post types would be selected, force the default required one
+					if ( cleanPostType.length === 0 ) {
+						cleanPostType.push( defaultRequiredSlug );
+					}
 					onChange( cleanPostType );
```

Comment:
Nit: tweak variable name and comment to clarify that this component can be used for data types other than post types, too (since it's used for the "additional post statuses" panel as well):

```suggestion
					const cleanOptions = [ ...new Set( values ) ];
					if ( value && cleanOptions.indexOf( slug ) === -1 ) {
						cleanOptions.push( slug );
					} else if ( ! value && cleanOptions.indexOf( slug ) > -1 ) {
						cleanOptions.splice( cleanOptions.indexOf( slug ), 1 );
					}
					// If no options would be selected, force the default required one
					if ( defaultRequiredSlug && cleanOptions.length === 0 ) {
						cleanOptions.push( defaultRequiredSlug );
					}
					onChange( cleanOptions );
```